elasticdl/python/data/reader/text_reader.py

```diff
@@ -0,0 +1,83 @@
+# Copyright 2020 The ElasticDL Authors. All rights reserved.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import csv
+import linecache
+
+import numpy as np
+import tensorflow as tf
+
+from elasticdl.python.data.reader.data_reader import (
+    AbstractDataReader,
+    Metadata,
+)
+
+
+class TextDataReader(AbstractDataReader):
+    """This reader is used to create shards for a file and
+    read records from the shard.
+    """
+
+    def __init__(self, **kwargs):
+        """
+        Args:
+            kwargs should contains "filename" and "records_per_task".
+        """
+        AbstractDataReader.__init__(self, **kwargs)
+        self._kwargs = kwargs
+        self._filename = self._kwargs["filename"]
+
+    def read_records(self, task, shuffle=False):
+        records = linecache.getlines(task.shard.name)[
+            task.shard.start : task.shard.end
+        ]
+        if shuffle:
+            np.random.shuffle(records)
+
+        for record in records:
+            yield record
+
+    def create_shards(self):
+        shard_name_prefix = self._kwargs["filename"] + ":shard_"
```

Comment:
What if the csv file has headers?

Reply:
There are many limits if we use a CSV file to store the meta information of images. Users need to pay attention to the header to create dataset if the CSV file has a header line. I think we should give users some rules to create the CSV file.